### PR TITLE
api: get_tips implementation

### LINKED ISSUE #426: getTips implementation

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #426 

# Test Plan:
Added tests

### CODE REVIEW

File: cclient/response/get_tips.c [hunk 1]
```diff
@@ -7,14 +7,39 @@
...
-get_tips_res_t* get_tips_res_new() { return flex_hash_array_new(); }
+get_tips_res_t* get_tips_res_new() {
+  get_tips_res_t* res = (get_tips_res_t*)malloc(sizeof(get_tips_res_t));
+  if (res) {
+    res->tips = flex_hash_array_new();
+  }
+  return res;
+}
+
+void get_tips_res_free(get_tips_res_t** const res) {
```
Comment (th0br0) [inline cclient/response/get_tips.c]: Could we start with #494 here? :D

File: cclient/response/get_tips.c [hunk 2]
```diff
@@ -7,14 +7,39 @@
...
+    flex_hash_array_free(tmp->tips);
+  }
+  free(tmp);
+  *res = NULL;
 }
 
-int get_tips_res_hash_num(get_tips_res_t* hashes) {
-  return flex_hash_array_count(hashes);
+get_tips_res_t* get_tips_res_add_tip(get_tips_res_t* const res,
+                                     tryte_t const* const tip) {
```
Comment (th0br0) [inline cclient/response/get_tips.c:30]: We should take `flex_trit` here and only serialize to string when converting to JSON

Comment (th0br0) [reply cclient/response/get_tips.c:30]: e.g. see below where we erroneously convert tryte_t to trit_array_p all of a sudden

**Review by th0br0 — CHANGES_REQUESTED** [review]
Guess this depends on #487

File: cclient/response/get_tips.c (hunk 2 above)
Comment (thibault-martinez) [reply cclient/response/get_tips.c:30]: @oopsmonk  is doing it in #473

**Review by th0br0 — APPROVED** [review]

**Pull request merged**